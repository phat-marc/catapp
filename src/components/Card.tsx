import * as React from 'react';
import { ICat } from '../containers/App';

const Card: React.SFC<ICat> = ({ id, name, email }) => {
	// destructuring works like this or just do it direct above instead of props
	// const { id, name, email } = props;
	return (
		<div className='catcard tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>	
			<img alt='robot profile' src={`https://robohash.org/${id}?set=set4&size=150x150`} />
			<div>
				<h2>{name}</h2>
				<p>{email}</p>
			</div>
		</div>
	);
}

export default Card;