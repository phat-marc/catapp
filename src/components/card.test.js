// import { shallow, mount, render } from 'enzyme';
import { shallow } from 'enzyme';
import React from 'react';
import Card from './card.js';
// import renderer from 'react-test-renderer';

// console.log(shallow(<Card />));
// it('expect to render card component', () => {
// 	expect(shallow(<Card />).length).toEqual(1)
// })
// this is a silly test but shows that we can render card component on its own


//snaphot - udemy 
describe('cards', () => {
	it('expect to render card component', () => {
		expect(shallow(<Card />)).toMatchSnapshot()
	})
})
// this creates a folder called snapshots

//snaphot - from jest documentation ***THIS IS not THE WAY***
// it('expect to render card component', () => {
// 	const tree = renderer
// 		.create(<Card />)
// 		.toJSON();
// 	expect(tree).toMatchSnapshot();
// })

// to update 
// 	$ jest --updateSnapshot