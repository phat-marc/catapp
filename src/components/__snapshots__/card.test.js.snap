// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card component 1`] = `ShallowWrapper {}`;
